internal/proxy/oauthproxy.go

```diff
@@ -131,11 +132,26 @@ func (u *UpstreamProxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 // upstreamTransport is used to ensure that upstreams cannot override the
 // security headers applied by sso_proxy
-type upstreamTransport struct{}
+type upstreamTransport struct {
+	InsecureSkipVerify bool
+}
 
 // RoundTrip round trips the request and deletes security headers before returning the response.
 func (t *upstreamTransport) RoundTrip(req *http.Request) (*http.Response, error) {
-	resp, err := http.DefaultTransport.RoundTrip(req)
+	transport := &http.Transport{
```

Comment:
Rather than creating a new transport every time, can we make this initialize one once and reuse it instead? This will hopefully reduce the overhead of TCP connections made.